examples/config.yaml

```diff
@@ -0,0 +1,162 @@
+# This demonstrates a file configuration that falls out naturally from the SDK configuration options defined in the specification.
+# The configuration shown is a "kitchen sink" example, with all the available options shown for demonstration purposes.
+
+exporters:
+  otlp/exporterA:
+    protocols:
+      grpc:
+        endpoint: http://localost:4317
+        headers:
+          # TODO: Replace with environment variable when parsing to avoid storing secret in plain text
+          api-key: 1234
+        compression: gzip
+        timeout_millis: 30_000
+  logging:
+
+# The resource, can be used across providers
+resources:
+  resource/resourceA:
+    # List of enabled resource detectors. Each detector has a name (FQCN for java), and an optional list of excluded keys. Detector name supports wildcard syntax. Detectors are invoked and merged sequentially. Static attributes and schema_url comprise a resource which is merged last.
+    detectors:
+      - name: com.domain.resources.CustomResourceProvider
+      - name: io.opentelemetry.sdk.extension.resources.*
+        excluded_attribute_keys:
+          # Exclude process.command_line, which is often verbose and may contain secrets
+          - process.command_line
+    # List of static resource attribute key / value pairs.
+    attributes:
+      service.name: my-service
+      service.instance.id: 1234
+    # The resource schema URL.
+    schema_url: http://schema.com
+
+# Limits
+limits:
```

Comment:
Do you imagine resources / limits applying to configuration outside the sdk block? If not, shouldn't they be nested under SDK? Alternatively, maybe all the top level fields of the sdk block can be promoted to the top level.

For some context, I originally nested everything under the sdk block because I imagined the scope expanding to configuration of instrumentation later. We currently don't have any examples of instrumentation configuration that is cross-language so its hard to imagine what that might entail, but those options felt distinct from SDK configuration.

Reply:
> Alternatively, maybe all the top level fields of the sdk block can be promoted to the top level

I would be ok with this idea. It would probably be a good idea to capture some example instrumentation/language specific options in here to get a sense for what that looks like.

Reply:
An example that could be used is in Erlang has a set of options for the "span sweeper" we have in the SDK:

```
{sweeper, #{interval => integer(),
           strategy => end_span | drop,
           span_ttl => integer(),
           storage_size => integer()}}
```

https://github.com/open-telemetry/opentelemetry-erlang/tree/main/apps/opentelemetry#span-sweeper

Been meaning to update the environment variables to `OTEL_ERLANG_*`, but then have to support both for backwards compatibility :(

Reply:
Opened #8 to track. I'm going to go through the otel java agent and try to compile some examples.